README.md

```diff
@@ -1,2 +1,101 @@
-# linkerd2-conformance
-A Conformance test suite to make sure Linkerd2 works well in different Kubernetes environments with different configurations.
+![Linkerd][logo]
+
+This repo contains the conformance tests for [Linkerd2](https://github.com/linkerd/linkerd2) as described by [this](https://github.com/linkerd/rfc/pull/24) RFC.
+
+The conformance validation tool is primarily intended to be run against an installation of Linkerd2 to verify the correctness of a [Kubernetes](https://kubernetes.io/) cluster's configuration with respect to Linkerd as well as validate non-trivial network communication (HTTP, gRPC, websocket) among stateless and stateful workloads in the Linkerd data plane.
+
+The conformance tests exercise the following features:
+
+- [ ] Validation of your LInkerd2 control plane
+- [ ] Automatic proxy injection on workloads
+- [ ] Functioning of  `linkerd tap`, `stat`, `routes` and `edges` commands
+- [ ] Verifying the functioning of the `tap` extension API server
+- [ ] Retries and timeouts
+- [ ] Data plane health checks
+- [ ] Ingress configuration
+
+_...and a lot more to come_
+
+The Linkerd project is hosted by the Cloud Native Computing Foundation ([CNCF](https://www.cncf.io/)).
+
+## Usage
+
+This section outlines the various methods that can be used to run the conformance tests against your [Kubernetes](https://kubernetes.io/) cluster
+
+### Using Sonobuoy
+
+[Sonobuoy](https://github.com/vmware-tanzu/sonobuoy) offers a reliable way to run diagnostic tests in a Kuberenetes cluster. We leverage its [plugin model](https://sonobuoy.io/docs/master/plugins/) to run conformance tests inside a Kubernetes pod.
+
+#### Using the Sonobuoy CLI
+
+These commands assume that the user has the [Sonobuoy CLI](https://github.com/vmware-tanzu/sonobuoy#installation) and [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) (with a correctly configured `kubeconfig`) installed locally.
+
+This repo provides a Sonobuoy plugin file that is intended to be plugged into the Sonobuoy CLI. Sonobuoy reads the plugin definition, and spins up a pod with the [linkerd2-conformance Docker image]().
+
+```bash
+# Download the Sonobuoy plugin file
+$ curl -LJO https://raw.githubusercontent.com/mayankshah1607/linkerd2-conformance/master/sonobuoy/plugin.yaml
+
+# Run the plugin
+$ sonobuoy run --plugin plugin.yaml --wait
```

Comment:
Looks like `--plugin` can take a URL directly.

Reply:
Yep! Downloading the plugin gives users the freedom to set their own options in the plugin definition. For example, the start up command as defined in the plugin currently uses a set of flags. Users could append flags like `--ginkgo.progress` to show progress reports on the pod logs, or `-linkerd-namespace` to modify the control plane ns before kicking off sonobuoy.

But for folks who want to use it directly out of the box, we should document that the plugin URL can be used directly! :)